examples/tutorials/advanced/cartesian_histograms.py

```diff
@@ -0,0 +1,340 @@
+r"""
+Cartesian histograms
+====================
+
+Cartesian histograms can be generated using the :meth:`pygmt.Figure.histogram`
+method. In this tutorial different histogram related aspects are addressed:
+
+- using vertical and horizontal bars
+- showing counts and frequency percent
+- adding annotations to the bars
+- showing cumulative values
+- using color and pattern as fill for the bars
+- using overlaid, stacked, and grouped bars
+"""
+
+# sphinx_gallery_thumbnail_number = 4
+
+
+# Import the required packages
+import numpy as np
+import pygmt
+
+###############################################################################
+# Generate random data from a normal distribution
+
+np.random.seed(100)
+
+# Mean of distribution
+mean = 100
+# Standard deviation of distribution
+stddev = 20
+
+# Create two data sets
+data01 = mean + stddev * np.random.randn(42)
+data02 = mean + stddev * 2 * np.random.randn(42)
+
+
+###############################################################################
+# Vertical and horizontal bars
+# ----------------------------
+# To define the width of the bins the ``series`` parameter has to be specified.
+# The bars can be filled via the ``fill`` parameter with either a color or a
+# pattern (see later in this tutorial). Use the ``pen`` parameter to adjust the
+# outline of the bars.
+
+# Create new figure instance
+fig = pygmt.Figure()
+
+# Create histogram for data01
+fig.histogram(
+    region=[0, 200, 0, 10],  # xmin, xmax, ymin, ymax
```

Comment:
Usually, we don't know the range of the dataset unless we do some statistics before plotting, especially *ymax* which highly depends on the choice of bin width.

There are two ways to let GMT automatically determine the region parameter.

1. Do not specify the `region` parameter
2. Use ymax=0, i.e., `region=[0, 200, 0, 0]`